text-to-speech/src/main/java/com/ibm/watson/developer_cloud/text_to_speech/v1/TextToSpeech.java

```diff
@@ -275,6 +274,26 @@ public TextToSpeech(String username, String password) {
     final boolean isNew = model.getId() == null;
     final String path = isNew ? PATH_CUSTOMIZATIONS : String.format(PATH_CUSTOMIZATION, model.getId());
 
+    // For update, a specified language cannot differ from existing model.
+    if (!isNew && model.getLanguage() != null) {
+      final Request request = RequestBuilder.get(path).build();
+      createServiceCall(request, new ResponseConverter<CustomVoiceModel>() {
+
+        // The existing model can be null for unit test.
+        @Override
+        public CustomVoiceModel convert(Response response) {
+          CustomVoiceModel existing = ResponseUtils.getObject(response, CustomVoiceModel.class);
+
+          if (existing != null) {
+            Validator.isTrue(model.getLanguage().equals(existing.getLanguage()),
+                             "cannot change language of existing model");
+          }
+
+          return null;
+        }
+      }).execute();
+    }
```

Comment:
@germanattanasio the service doesn't provide a means of changing the language of an existing model, so there's no equivalent error that we can return.  With the service API, once you create a model, its language is immutable.  The problem with the Java SDK is the combination of the create and update functions into a single method.  The best that we can do is prevent the attempt to change the language of an existing model to avoid problems.  I allow the user to pass the same language for a model, since that's not a problem and doesn't require that they remove the language from a local copy of the model before using it as the argument for an update request; it's only a language change that fails.

The best solution is to create separate methods for the create and update operations to mimic the service and prevent the possibility altogether.  But I first wanted to be sure that the SDK's existing method doesn't allow the unsupported operation.